gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-debug/src/main/java/io/gravitee/gateway/debug/policy/impl/DebugPolicy.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.debug.policy.impl;
+
+import io.gravitee.gateway.api.ExecutionContext;
+import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.gateway.api.stream.ReadWriteStream;
+import io.gravitee.gateway.policy.Policy;
+import io.gravitee.gateway.policy.PolicyException;
+import io.gravitee.gateway.policy.StreamType;
+import io.gravitee.policy.api.PolicyChain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class DebugPolicy implements Policy {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DebugPolicy.class);
+    private final StreamType streamType;
+    private final Policy policy;
+
+    public DebugPolicy(StreamType streamType, Policy policy) {
+        this.streamType = streamType;
+        this.policy = policy;
+    }
+
+    @Override
+    public String id() {
+        return policy.id();
+    }
+
+    @Override
+    public void execute(PolicyChain chain, ExecutionContext context) throws PolicyException {
+        LOGGER.debug("Before execute() for {}", streamType);
+        policy.execute(chain, context);
+    }
+
+    @Override
+    public ReadWriteStream<Buffer> stream(PolicyChain chain, ExecutionContext context) throws PolicyException {
+        LOGGER.debug("Before stream() for {}", streamType);
```

Comment:
IMHO, this log doesn't make sense as it sounds like a log for announcing "method entering".

Reply:
removed